Updating `fc_export()`

kenkomodo opened the pull request:
Closes #16 

Updates:
- Added `ragg` as conditional dependency
  - `ragg` used for png, jpeg, and tiff formats when user has it installed; if not installed, function falls back on `grDevices` and outputs a message to the user to alert them that `ragg` may improve their image quality. _Note_: `ragg` does not have an analogous function to generate bmp files, so bmp will always use `grDevices`
- Added options for vector formats (svg, pdf), ensuring that if other units are supplied when using these formats they are converted to inches (underlying functions only accept units in inches)
- Updated function documentation, vignette, and NEWS.md accordingly

pasahe commented on R/fc_export.R:
I love your implementation! Just a few comments:

- I think that the warning about the default units in inches when using 'svg' or 'pdf' format shouldn't be printed if no unit is specified. So I would change the if statement in lines 99 and 106 from `units != "in"` to `!is.na(units) & units != "in"`. 

- Since the `res` parameter is not used for vector formats, I think it would be nice to make it explicit in the description of the `res` parameter and add a warning if the resolution is specified and a vector format is chosen.

pasahe reviewed (changes requested):
Please consider my comments on the units and resolution before merging it.

Thank you!!

kenkomodo commented:
> I love your implementation! Just a few comments:
> 
> * I think that the warning about the default units in inches when using 'svg' or 'pdf' format shouldn't be printed if no unit is specified. So I would change the if statement in lines 99 and 106 from `units != "in"` to `!is.na(units) & units != "in"`.
> * Since the `res` parameter is not used for vector formats, I think it would be nice to make it explicit in the description of the `res` parameter and add a warning if the resolution is specified and a vector format is chosen.

@pasahe Thanks for your review and suggestions. The reason the warning still got thrown from your first point is because the default units were set to `"px"` for the function. I've updated the function so that the default is `NULL` - in which case, the function will select `"px"` as the bitmap default and `"in"` as the vector default.

I've updated to include the explicit mention in the description and I also added a warning in to make sure that if the user supplies `units`, what they specify must be a valid option for their specified format.

Cheers!

pasahe reviewed (approved):
Now it's perfect. Thank you so much!
